lms/lmsweb/views.py

```diff
@@ -236,6 +226,36 @@ def read_all_notification():
     return jsonify({'success': success_state})
 
 
+@webapp.route('/share', methods=['POST'])
+@login_required
+def share():
+    act = request.json.get('act')
+    solution_id = int(request.json.get('solutionId', 0))
+    solution, shared_solution = share_link.get(solution_id)
+
+    if act == 'get':
+        if shared_solution is None:
```

Comment:
add before: `shared_url = shared_solution.shared_url`

remove the jsonify inside the if